### ISSUE TITLE
Option to conditionally log to stderr only above a given level

### ISSUE THREAD

**Issue by abonander** [rust-cli/env_logger#219]
When using `env_logger` in a binary running in Google Kubernetes Engine, everything it logs gets recorded as an error log because it logs to `stderr` by default:

![image](https://user-images.githubusercontent.com/3198595/137211196-a0c58b9a-160e-4dbb-8161-b6d3493007d4.png)

This is just the actix-web logger middleware logging requests from the Gcloud healthcheck service at `INFO` (plus an easter egg of a web crawler looking for an admin login form?), but Gcloud thinks they're error logs because they were printed to `stderr`.

Of course, we could just set `Target::Stdout` and call it a day, but I _would_ like to retain the ability to print actual error logs to `stderr` as we want to set this up to page someone if our API service is throwing actual errors.

I'm thinking another variant to `Target` that looks something like:

```rust
enum Target {
    // ...

    // Print to `stderr` if the record level is at or above the given `log::Level`, otherwise `stdout`.
    StdoutOrStderrAt(log::Level),
}
```

I'm also considering, as a workaround, using the `Builder::format()` method to conditionally print to stderr but I'm not sure if this would break things. It's suboptimal either way:

```rust
builder.format(|f, record| {
    if record.level() >= log::Level::Error {
        eprintln!(/* format `record` using the same format as the default formatter */)
    } else {
        write!(fmt, "{}", record.args())
    }
})
```

**Comment by Zarathustra2** [comment]
Hey @abonander we are also running into this issue, what did you end up doing?

**Comment by abonander** [comment]
I think we just changed the default sink to stdout so it stopped logging everything as errors.

**Comment by Zarathustra2** [comment]
I had a helpful discussion with people on the rust irc, will likely go with this: https://cloud.google.com/logging/docs/structured-logging and make it so that it picks up the correct severity! Thanks for the input though! :)

**Comment by bonigarcia** [comment]
Any update on this? IMO, it would be interesting to redirect the `Level::Error` messages to stderr and the rest levels to stdout.

**Comment by jplatte** [comment]
No, this crate is actively maintained. I did a release not long ago to fix the most glaring issues, but don't intend to spend more time on it and doesn't seem like any of the other people with access rights want to, either. See #159

**Comment by epage** [comment]
WG-CLI is picking up passive maintenance of this script.

I'm not familiar with that technology stack but this overall feels like a problem with how output is interacting with the technology stack.

**Comment by mercuriete** [comment]
I am having the same problem.
meilisearch is having the same problem but they tell me its not their fault:
https://github.com/meilisearch/meilisearch/discussions/2514

Please do the following.
By default, you should output to 2 different files.
stdout: when the log is in log level below error.
stderr: when the log is equal or above error.

changing the default behaviour should be good enough to make happy to all parts.

Thanks in advance.
